students/YingGuo/session02/GridPrinter.py

```diff
@@ -0,0 +1,87 @@
+
+def print_function1():
+    """homework2 #1, print a grid as shown in homework"""
+    print("+ - - - - + - - - - +")
+    for i in range(4):
+        print("|         |         |")
+    print("+ - - - - + - - - - +")
+    for i in range(4):
+        print("|         |         |")
+    print("+ - - - - + - - - - +")
+    return
+
+
+
+def print_grid(n):
+    """make the print grid function more general.
+    This is a function print_grid(n) that takes one integer argument and prints a grid just like before,
+    BUT the size of the grid is given by the argument,which is n"""
+    for x in range(n+2):
+
+        if x == 0:
+            for i in range(n+2):
+                if i == 0:
+                    print("+", end=" ")
+                elif i == (n+2)//2:
+                    print("+", end=" ")
+                elif i == n+1:
+                    print("+", end=" ")
+                else:
+                    print("-", end=" ")
+            print()
+        
+        elif x == (n+2)//2:
+            for i in range(n+2):
+                if i == 0:
+                    print("+", end=" ")
+                elif i == (n+2)//2:
+                    print("+", end=" ")
+                elif i == n+1:
+                    print("+", end=" ")
+                else:
+                    print("-", end=" ")
+            print()
+
+        elif x == n+1:
+            for i in range(n+2):
+                if i == 0:
+                    print("+", end=" ")
+                elif i == (n+2)//2:
+                    print("+", end=" ")
+                elif i == n+1:
+                    print("+", end=" ")
+                else:
+                    print("-", end=" ")
+            print()
+        
+        else:
+            for i in range(n+2):
+                if i == 0:
+                    print("|", end=" ")
+                elif i == (n+2)//2:
+                    print("|", end=" ")
+                elif i == n+1:
+                    print("|", end=" ")
+                else:
+                    print(" ", end=" ")
+            print()
```

Comment:
There's a lot of repeated code in this function, and it'd be desirable to avoid such repetition as much as possible. It looks like you achieved that in the next part, so I'd say it's OK here, but always remember the DRY (Don't Repeat Yourself) principle (though it might not make much sense now, hopefully it becomes more evident as we go along).